shared/sbl.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+type TypeFilter = (domain: string, selector: string, data: Array<*>) => ?boolean
+type TypeDomain = {
+  selectors: {[string]: Function},
+  domainFilters: Array<TypeFilter>,
+  selectorFilters: {[string]: TypeFilter},
+  data: Object
+}
+
+var domains: {[string]: TypeDomain} = {}
+var globalFilters: Array<TypeFilter> = []
+var pendingCalls = {} // allows sbl to be called regardless of module load order
+
+const DOMAIN_SELECTOR_REGEX = /^([^/]+)(.+)/
+
+function sbl (path: string, ...data?: Array<*>) {
+  var [, name, selector] = DOMAIN_SELECTOR_REGEX.exec(path)
+  var domain = domains[name]
+  // Filters can perform additional functions, and by returning `false` they
```

Comment:
Its not a error specific to the problem which is helpful since we can check it when it occurs. If a truthy exist is overhead then we have some sins to answer for in the Inbox vue that are more sinister.